akka-persistence/src/main/scala/akka/persistence/journal/PersistencePluginProxy.scala

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.journal
+
+import java.net.URISyntaxException
+import java.util.concurrent.TimeoutException
+
+import akka.actor._
+import akka.persistence.{ AtomicWrite, DeleteMessagesFailure, DeleteSnapshotFailure, DeleteSnapshotsFailure, JournalProtocol, NonPersistentRepr, Persistence, SaveSnapshotFailure, SnapshotProtocol }
+import akka.util.Helpers.Requiring
+import com.typesafe.config.Config
+
+import scala.concurrent.duration._
+
+object PersistencePluginProxy {
+  final case class TargetLocation(address: Address)
+  private case object InitTimeout
+
+  def setTargetLocation(system: ActorSystem, address: Address): Unit = {
+    Persistence(system).journalFor(null) ! TargetLocation(address)
+    if (system.settings.config.getString("akka.persistence.snapshot-store.plugin") != "")
+      Persistence(system).snapshotStoreFor(null) ! TargetLocation(address)
+  }
+
+  def start(system: ActorSystem): Unit = {
```

Comment:
One thought. We could make this an Akka [Extension](http://doc.akka.io/docs/akka/2.4.0/scala/extending-akka.html) instead. Then those methods would not need to have a `ActorSystem` parameter and it is possible to configure an extension to be loaded at system startup, as alternative to calling this start() method.

In the extension we would do `Persistence(system).journalFor(null)` in the constructor of the extension.

WDYT?